pkg/controller/jobframework/validation.go

```diff
@@ -161,29 +163,43 @@ func validateUpdateForMaxExecTime(oldJob, newJob GenericJob) field.ErrorList {
 	return nil
 }
 
-// ValidateImmutablePodSpec function is used for serving workloads to ensure no changes are allowed
-// to the PodSpec except for the image field in containers.
-func ValidateImmutablePodSpec(newPodSpec *corev1.PodSpec, oldPodSpec *corev1.PodSpec, fieldPath *field.Path) field.ErrorList {
-	// handle updateable fields by munging those fields prior to deep equal comparison.
-	mungedPodSpec := newPodSpec.DeepCopy()
-
-	// munge spec.containers[*].image
-	newContainers := make([]corev1.Container, 0, len(newPodSpec.Containers))
-	for i, container := range mungedPodSpec.Containers {
-		container.Image = oldPodSpec.Containers[i].Image
-		newContainers = append(newContainers, container)
-	}
-	mungedPodSpec.Containers = newContainers
+// ValidateImmutablePodGroupPodSpec function is used for serving workloads to ensure no changes are allowed
+// to the PodSpec except fields that required for role-hash generation.
+func ValidateImmutablePodGroupPodSpec(newPodSpec corev1.PodSpec, oldPodSpec corev1.PodSpec, fieldPath *field.Path) field.ErrorList {
```

Comment:
I think it is better to pass the Specs as pointers here to avoid creating copies of them, especially in webhooks which are performance-sensitive.

Reply:
+1

Reply:
Done